tests/test_catalog.py

```diff
@@ -231,39 +231,59 @@ def test_should_download_available_quicklooks(self, requests_mock: req_mock.Mock
         )
         assert out_paths == [str(tmp_path / f"quicklook_{image_id}.jpg")]
 
-
-def test_construct_search_parameters(catalog_mock):
-    assert (
-        catalog_mock.construct_search_parameters(
-            geometry=SIMPLE_BOX,
-            collections=[PHR],
-            start_date="2014-01-01",
-            end_date="2022-12-31",
-            usage_type=["DATA", "ANALYTICS"],
-            limit=4,
-            max_cloudcover=20,
-        )
-        == SEARCH_PARAMETERS
+    @pytest.mark.parametrize(
+        "usage_type",
+        [
+            {"usage_type": ["DATA"]},
+            {"usage_type": ["ANALYTICS"]},
+            {"usage_type": ["DATA", "ANALYTICS"]},
+            {},
+        ],
+        ids=[
+            "usage_type:DATA",
+            "usage_type:ANALYTICS",
+            "usage_type:DATA, ANALYTICS",
+            "usage_type:None",
+        ],
     )
-
-
-def test_construct_search_parameters_fc_multiple_features_raises(catalog_mock):
-    with open(
-        pathlib.Path(__file__).resolve().parent / "mock_data/search_footprints.geojson",
-        encoding="utf-8",
-    ) as file:
-        fc = json.load(file)
-
-    with pytest.raises(ValueError) as e:
-        catalog_mock.construct_search_parameters(
-            geometry=fc,
-            start_date="2020-01-01",
-            end_date="2020-08-10",
-            collections=[PHR],
-            limit=10,
-            max_cloudcover=15,
-        )
-    assert str(e.value) == "UP42 only accepts single geometries, the provided geometry contains multiple geometries."
+    @pytest.mark.parametrize(
+        "max_cloudcover",
+        [
+            {"max_cloudcover": 100},
+            {},
+        ],
+        ids=[
+            "max_cloudcover:100",
+            "max_cloudcover:None",
+        ],
+    )
+    def test_should_construct_search_parameters(self, usage_type: dict, max_cloudcover: dict):
+        params = {
+            "geometry": SIMPLE_BOX,
+            "collections": [PHR],
+            "start_date": START_DATE,
+            "end_date": END_DATE,
+            **usage_type,
+            **max_cloudcover,
+        }
+        response = {**SEARCH_PARAMETERS}
+        optional_response: dict = {"query": {}}
```

Comment:
Why do we need type hint here?

Reply:
`mypy` complains about it. I believe the assignment of the query dictionary makes it required. 
![Screenshot 2024-09-25 at 12 09 38](https://github.com/user-attachments/assets/e87e3ffd-60fa-497d-8af4-8a6205b7cecf)